setup.py

```diff
@@ -88,7 +88,8 @@ def get_extras_require() -> Dict[str, List[str]]:
             "bokeh<2.0.0",
             "chainer>=5.0.0",
             "cma",
-            "lightgbm",
+            "lightgbm>=2.2.2",
```

Comment:
I think this constraint is reasonable. This is required to support pandas categorical datatype. Pandas categorical datatype is useful and widely used to save memory usage.